fortran/micm.F90

```diff
@@ -51,23 +51,21 @@ subroutine delete_micm_c(micm, error) bind(C, name="DeleteMicm")
       type(error_t_c),    intent(inout) :: error
     end subroutine delete_micm_c
 
-    subroutine micm_solve_c(micm, time_step, temperature, pressure, air_density, num_concentrations, concentrations, &
-                num_user_defined_reaction_rates, user_defined_reaction_rates, solver_state, solver_stats, error) &
+    subroutine micm_solve_c(micm, time_step, temperature, pressure, air_density, concentrations, &
+                user_defined_reaction_rates, solver_state, solver_stats, error) &
                 bind(C, name="MicmSolve")
       use musica_util, only: string_t_c, error_t_c
       import c_ptr, c_double, c_int, solver_stats_t_c
-      type(c_ptr), value, intent(in)         :: micm
-      real(kind=c_double), value, intent(in) :: time_step
-      real(kind=c_double), value, intent(in) :: temperature
-      real(kind=c_double), value, intent(in) :: pressure
-      real(kind=c_double), value, intent(in) :: air_density
-      integer(kind=c_int), value, intent(in) :: num_concentrations
-      real(kind=c_double), intent(inout)     :: concentrations(num_concentrations)
-      integer(kind=c_int), value, intent(in) :: num_user_defined_reaction_rates
-      real(kind=c_double), intent(inout)     :: user_defined_reaction_rates(num_user_defined_reaction_rates)
-      type(string_t_c), intent(out)          :: solver_state
-      type(solver_stats_t_c), intent(out)    :: solver_stats
-      type(error_t_c), intent(inout)         :: error
+      type(c_ptr),         value, intent(in)    :: micm
+      real(kind=c_double), value, intent(in)    :: time_step
+      type(c_ptr),         value, intent(in)    :: temperature
+      type(c_ptr),         value, intent(in)    :: pressure
+      type(c_ptr),         value, intent(in)    :: air_density
+      type(c_ptr),         value, intent(in)    :: concentrations
+      type(c_ptr),         value, intent(in)    :: user_defined_reaction_rates
+      type(string_t_c),           intent(out)   :: solver_state
```

Comment:
Doesn't `concentrations` have to be `intent(inout)`?
```fortran
type(c_ptr), value, intent(inout) :: concentrations
```

Reply:
oh it's just `type(c_ptr)`, so probably not

Reply:
Yeah, I think this is just saying the address pointed to doesn't change